### ISSUE TITLE
cucumber-expressions: parameter type name with special characters not working 

### ISSUE THREAD

**Issue by lenntt** [cucumber/cucumber#387]
## Summary
Not all characters are properly escaped in js cucumber-expressions parameter types. For sure `[]`, but I think I've seen `.` also gone wrong..

## Steps to Reproduce
Given I've defined a type in 'cucumber-js'
```
defineParameterType( {
    name: '[string]',
    regexp: /".*"/,
    transformer: function(s) {   
        return s: // transformer is irrelevant  
   }  
} );
```
When I use it in a step definition:
```
When( 'My string list is {[string]}', function( _titles ) { ... } );
```
And I run my cukes

## Expected Behavior
Then it just works

## Current Behavior

Then it fails with a stacktrace:
```
Error: Undefined parameter type {\[string]}
    at UndefinedParameterTypeError.Error (native)
    at UndefinedParameterTypeError.CucumberExpressionError (./node_modules/cucumber-expressions/dist/src/errors.js:17:131)
    at new UndefinedParameterTypeError (./node_modules/cucumber-expressions/dist/src/errors.js:67:139)
    at ./node_modules/cucumber-expressions/dist/src/cucumber_expression.js:71:35
```

## Possible Solution
Would need to investigate

## Context & Motivation
I've a transformer that parses `"foo", "bar" and "baz"` into `["foo", "bar", "baz"]`.
I thought of `[string]` as a logical type name.

## Your Environment

* Version used: 4.2.1

**Comment by aslakhellesoy** [comment]
To keep things simple I think we should require Parameter type names to be `/\w+/`. No special characters allowed.

**Comment by lenntt** [comment]
Works for me,

But there are currently no (or little?) limitations, so consider this to be breaking. 
For example `string-list` now works, but doesnt pass `/\w+/`.

**Comment by aslakhellesoy** [comment]
Fixed by #410
